### PR TITLE
My Site Dashboard: Remove A/B experiment

### LINKED ISSUE #19133: My Site Dashboard: Remove default tab AB experiment
Remove AB test from code, and manually assign default tab based on application.

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Closes #19133

## Description
- This removed the My Site Dashboard A/B experiment completely from the code.
- Now the default tab will depend on the app:
    - The default tab will be Home for the Jetpack app.
    - The default tab will be Menu for the WordPress app.
- Old default tab assignments will not be overridden by this change.
- The user can still change their default tab from App Settings

### Jetpack - Old users

1. Checkout `trunk`
2. Update `BlogDashboardAB.swift` line 26 to return `.control` 
3. Do a fresh install
4. Login
5. Check that the default section is Menu
6. Checkout this branch
7. Install the app
8. Make sure that the default section is still Menu

 ### Jetpack - New users

1. Do a fresh install
2. Login
3. Make sure that the default section is Home
4. Go to App Settings -> Initial Screen and change the section to Menu
5. Close the app and open it again
6. Make sure that the default section is now Menu

### WordPress - Old users

1. Checkout `trunk`
2. Update `BlogDashboardAB.swift` line 26 to return `.treatment` 
3. Do a fresh install
4. Login
5. Check that the default section is Home
6. Checkout this branch
7. Install the app
8. Make sure that the default section is still Home

 ### WordPress - New users

1. Do a fresh install
2. Login
3. Make sure that the default section is Menu
4. Go to App Settings -> Initial Screen and change the section to Home
5. Close the app and open it again
6. Make sure that the default section is now Home

### Analytics

1. Run the app
2. Do any action
3. Check the logs and make sure the `default_tab_experiment` property is not added to events
4. Go to App Settings -> Initial Screen and change the section to Home
5. Make sure the event `app_settings_initial_screen_changed` is fired

## Regression Notes
1. Potential unintended areas of impact
N/A
3. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A
4. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by momo-ozawa — COMMENTED** [review]
Works as described!

This is unrleated to the changes in this PR, but I noticed that `app_settings_initial_screen_changed` is fired every time you tap an item, even if your actual selection doesn't change (i.e. "Menu" is selected then you tap "Menu" again). WDYT of only firing `app_settings_initial_screen_changed` when the selection changes?

**Comment by hassaanelgarem** [comment]
> WDYT of only firing app_settings_initial_screen_changed when the selection changes?

@momo-ozawa Great catch! Addressed in a7dd9e5b698e3fb6e47f738bb539e0183bed9f9b 👍

File: WordPress/Classes/ViewRelated/Me/App Settings/AppSettingsViewController.swift [hunk 1]
```diff
@@ -343,13 +343,15 @@ class AppSettingsViewController: UITableViewController {
             let viewController = SettingsSelectionViewController(dictionary: settingsSelectionConfiguration)
 
             viewController?.onItemSelected = { (section: Any!) -> () in
+                let oldDefaultSection = MySiteSettings().defaultSection
                 guard let section = section as? Int,
-                    let defaultSection = MySiteViewController.Section(rawValue: section) else {
+                      let newDefaultSection = MySiteViewController.Section(rawValue: section),
+                        newDefaultSection != oldDefaultSection  else {
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Me/App Settings/AppSettingsViewController.swift]: [nit] alignment and extra space before `else`
```suggestion
                      let newDefaultSection = MySiteViewController.Section(rawValue: section),
                      newDefaultSection != oldDefaultSection else {
```

**Review by momo-ozawa — APPROVED** [review]
LGTM w one tiny nit 🎉 

Confirming that `app_settings_initial_screen_changed` is fired only when the selection changes. Thanks @hassaanelgarem!

**Pull request merged**